spanner/api/Spanner.Samples.Tests/SpannerFixture.cs

```diff
@@ -438,4 +465,40 @@ public async Task RefillMarketingBudgetsAsync(int firstAlbumBudget, int secondAl
             await cmd.ExecuteNonQueryAsync();
         }
     }
+
+    public async Task RunWithTemporaryDatabaseAsync(Func<string, Task> testFunction, params string[] extraStatements)
+        => await RunWithTemporaryDatabaseAsync(InstanceId, testFunction, extraStatements);
+
+    public async Task RunWithTemporaryDatabaseAsync(string instanceId, Func<string, Task> testFunction,
+        params string[] extraStatements)
+    {
+        var databaseId = $"my-db-{DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()}";
+        await RunWithTemporaryDatabaseAsync(instanceId, databaseId, testFunction, extraStatements);
+    }
+    
+    public async Task RunWithTemporaryDatabaseAsync(string instanceId, string databaseId, Func<string, Task> testFunction, params string[] extraStatements)
+    {
+        var databaseAdminClient = await DatabaseAdminClient.CreateAsync();
+        var operation = await databaseAdminClient.CreateDatabaseAsync(new CreateDatabaseRequest
+        {
+            ParentAsInstanceName = InstanceName.FromProjectInstance(ProjectId, instanceId),
+            CreateStatement = $"CREATE DATABASE `{databaseId}`",
+            ExtraStatements = { extraStatements },
+        });
+        var completedResponse = await operation.PollUntilCompletedAsync();
+        if (completedResponse.IsFaulted)
+        {
+            throw completedResponse.Exception;
+        }
+
+        try
+        {
+            await testFunction(databaseId);
+        }
+        finally
+        {
+            // Cleanup the test database.
+            await databaseAdminClient.DropDatabaseAsync(DatabaseName.FormatProjectInstanceDatabase(ProjectId, instanceId, databaseId));
```

Comment:
I see the added value for the above for resources that are used for multiple tests, but I don't think that it would be possible for this kind of temporary test databases. There is a hard limit of max 100 databases per instance, and as the cleanup is only done during startup (and only deletes a limited number of resources), a small number of tests that use a temporary database could already cause a test run to hit that limit.
As noted above, the long initialization time and monolithic nature of the `SpannerFixture` is more a hassle than help for small, simple tests for features that do not involve slow processes (such as for example backups).